Reverse sync

sedge opened the pull request.

Plumpernickel committed:
Partial unit tests for HTTP routes

These were completely untested, and we needed this on the heels of a new refactor`

gideonthomas committed:
Add server side web socket interface for server->client sync

sedge committed:
Unit tests for socket connections

We had none! This also includes scaffolding and good examples of how to factor out common pieces of code.

gideonthomas commented on bower.json:
```diff
@@ -4,7 +4,9 @@
     "jquery": "2.1.1",
-    "makerstrap": "0.1.8"
+    "makerstrap": "0.1.8",
+    "superagent": "latest",
```
use a specific version

gideonthomas commented on bower.json:
```diff
@@ -4,7 +4,9 @@
-    "makerstrap": "0.1.8"
+    "makerstrap": "0.1.8",
+    "superagent": "latest",
+    "lodash": "latest"
```
here too

gideonthomas commented on client/src/comms.js:
```diff
@@ -0,0 +1,108 @@
+  }
+};
+
+
```
This is a WIP! Should probably just delete this file and I can add it in or something.

gideonthomas commented on server/lib/socket-server.js:
```diff
@@ -0,0 +1,47 @@
+var Sync = require( '../lib/sync'),
```
probably just want to do ./sync

gideonthomas commented on server/lib/socket-server.js:
```diff
@@ -0,0 +1,47 @@
+var Sync = require( '../lib/sync'),
+    SyncMessage = require('../lib/syncmessage'),
```
same here

gideonthomas commented on server/lib/sync.js:
```diff
@@ -6,7 +6,8 @@
     uuid = require( "node-uuid" ),
-    emitter = new ( require( "events" ).EventEmitter )();
+    emitter = new ( require( "events" ).EventEmitter )(),
+    SyncMessage = require( "../lib/syncmessage" );
```
./syncmessage

sedge committed:
Remove client code

This isn't ready for merging, but we want to keep it in history in the short-term.
